### PR TITLE
Address larger tech debt/TODOs

### PR DESCRIPTION
**Pull request by cee-chen**
## Summary

This addresses several misc tech debt TODOs I had across the repo. I strongly recommend following along by commits, since most are unrelated.

## QA

- [x] `yarn test:jest plugins/enterprise_search --coverage` within the `x-pack` dir passes with 100% coverage
- Manually smoke-test the following scenarios:
  - [x] Logging in as the `enterprise_search` user with App Search running on localhost:3002 returns an engines list as expected
  - [x] Logging in as the `elastic` user with App Search running on localhost:3002 returns "Cannot find App Search account" messaging
  - [x] Logging in as any user with App Search stopped/not running returns "Error Connecting to App Search" messaging

### CODE REVIEW

File: x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts [hunk 1]
```diff
@@ -9,38 +9,52 @@ import { schema } from '@kbn/config-schema';
...
+          });
+        } else {
+          // Either a completely incorrect Enterprise Search host URL was configured, or App Search is returning bad data
+          throw new Error(`Invalid data received from App Search: ${JSON.stringify(engines)}`);
+        }
+      } catch (e) {
+        log.error(`Cannot connect to App Search: ${e.toString()}`);
+        if (e instanceof Error) log.debug(e.stack);
+
+        return response.notFound({ body: 'cannot-connect' });
```
Comment (richkuz) [inline x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts:56]: Should we distinguish here between `cannot-connect` and some other malformed response error condition?

File: x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts [hunk 2]
```diff
@@ -9,38 +9,52 @@ import { schema } from '@kbn/config-schema';
...
+          headers: { Authorization: request.headers.authorization },
         });
-      }
 
-      const engines = await enginesResponse.json();
-      return response.ok({
-        body: engines,
-        headers: { 'content-type': 'application/json' },
-      });
+        if (enginesResponse.url.endsWith('/login')) {
```
Comment (richkuz) [inline x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts:33]: It's surprising to me that the `url` prop here would end with `/login` in case of an invalid account. Does App Search redirect to `/login` here when an invalid auth header is provided? Calling the API directly would return a `401` status code.

**Review by richkuz — APPROVED** [review]
I like the extra tests! None of my questions are blocking, just trying to learn more.

File: x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts (hunk 1 above)
Comment (cee-chen) [reply x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts:56]: We can - I can't think of anything super useful to show the user in that case though 🤔

I think the assumption that malformed data from App Search = 'This isn't App Search' is a relatively safe one in terms of end-user-experience, as we can't show them anything but an error message in that case anyway. Definitely open to other thoughts or suggestions however!

File: x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts (hunk 2 above)
Comment (cee-chen) [reply x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts:33]: Yep, this is on App Search's end - I believe our Ruby logic is to default to the /login page no matter what if we encounter any invalid auth (even when hitting API endpoints).

Comment (cee-chen) [reply x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts:33]: To be clear I don't 100% agree with it (as someone who prefers a very RESTful API/separate your front-end & back-end POV :), but I also would have no idea how to change it, so probably out of scope for now.

File: x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts (hunk 1 above)
Comment (richkuz) [reply x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts:56]: Good point -- not much we can show the user. The more adventurous user or support engineer can still glean valuable lower-level detail by poking into the stack trace that we're logging to the console. 👍

File: x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx [hunk 3]
```diff
@@ -49,12 +49,8 @@ export const EngineOverview: ReactFC<> = () => {
   };
   const hasValidData = response => {
     return (
-      response &&
-      Array.isArray(response.results) &&
-      response.meta &&
-      response.meta.page &&
-      typeof response.meta.page.total_results === 'number'
-    ); // TODO: Move to optional chaining once Prettier has been updated to support it
+      Array.isArray(response?.results) && typeof response?.meta?.page?.total_results === 'number'
```
Comment (JasonStoltz) [inline x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx]: Woah rad, I didn't realize we could start use this.

**Comment by JasonStoltz** [comment]
Good stuff 👍

**Review by JasonStoltz — APPROVED** [review]
